src/libs/search/utils/buildOfferGenreTypes.ts

```diff
@@ -0,0 +1,22 @@
+import { GenreType, GenreTypeContentModel } from 'api/gen'
+import { OfferGenreType } from 'features/search/types'
+import { GenreTypeMapping } from 'libs/subcategories/types'
+
+export const buildOfferGenreTypes = (
+  genreType: GenreType,
+  genreSubtypeNames: string[],
+  genreTypeMapping: GenreTypeMapping
+): OfferGenreType[] | undefined => {
+  if (!genreSubtypeNames) return
+  const genreTypeValues = genreTypeMapping[genreType]
+
+  const offerGenreTypes = genreSubtypeNames.map((genreSubtypeName: string) => {
+    const genreSubtypeItem = genreTypeValues.find(
+      (genreTypeValue: GenreTypeContentModel) => genreTypeValue.name === genreSubtypeName
+    )
+    if (!genreSubtypeItem) return
+    return { key: genreType, ...genreSubtypeItem }
+  })
+
+  return offerGenreTypes.filter(Boolean) as OfferGenreType[] // pas ouf le typage
```

Comment:
C'est vrai que c'est pas ouf le as ...
Est-ce que si au lieu de construire le OfferGenreType en faisant : 
`{ key: genreType, ...genreSubtypeItem }`
Tu fais plutôt : 
`{ key: genreType, name: genreSubtypeItem.name, value: genreSubtypeItem.value}`
typescript arrive à inférer que c'est un OfferGenreType ?

Reply:
En tout cas il faudra penser à supprimer le comm :)

Reply:
Je teste ça !

Reply:
ça ne fonctionne pas mieux, j'ai cette erreur 
`Type '({ key: GenreType; name: string; value: string; } | undefined)[]' is not assignable to type 'OfferGenreType[]'.
  Type '{ key: GenreType; name: string; value: string; } | undefined' is not assignable to type 'OfferGenreType'.
    Type 'undefined' is not assignable to type 'OfferGenreType'.
      Type 'undefined' is not assignable to type '{ key: GenreType; }`

Reply:
alors que le `.filter(Boolean)` devrait faire le taff pour supprimer les undefined de la liste, du coup je comprends pas

Reply:
Relou ... Est-ce que tu vuex qu'on essaye de regarder ensemble ou on laisse le 'as' ?

Reply:
J'avoue que j'aimerais boucler ça aujourd'hui, mais pourquoi pas regarder ensemble pour le prochain ticket qui réutilisera cette méthode

Reply:
okpm, ça peut être une tâche de créneau qualité le soir 👍